Client/Frontend/Settings/SettingsTableViewController.swift

```diff
@@ -136,6 +136,21 @@ private class PaddedSwitch: UIView {
     }
 }
 
+class WithAccountBoolSetting: BoolSetting {
+    unowned var settings: SettingsTableViewController
+
+    var profile: Profile {
+        return settings.profile
+    }
+
+    override var hidden: Bool { return !profile.hasAccount() }
```

Comment:
I don't think it's enough to hide or unhide.

If you have an FxA, but we haven't synced it yet — we don't have a populated meta/global — then we can't let you flip these settings. Who syncs the syncers?!